docs/semgrep-app/editor.md

```diff
@@ -0,0 +1,205 @@
+
+## Viewing rules
+
+![Screenshot of the Editor's main view composed of three panes and top menu](../img/editor-panes.png)
+
+The editor is composed of three panes and a top menu.
+
+<dl>
+    <dt>Library</dt>
+    <dd>View and open various rules through the library. It contains two top-level folders:
+        <ul>
+            <li>Your organization’s rules</li>
+            <li>Publicly-available Semgrep Registry rules from the <code>semgrep-rules</code> repository</li>
+        </ul>
+        <strong>Private rules</strong> are denoted by a <strong>lock</strong> icon overlaid over the file icon.
+    </dd>
+    <dt>YAML editor</dt>
+    <dd>Enter your rule’s YAML schema in this pane.</dd>
+    <dt>Sample code window</dt>
+    <dd>Enter your rule’s test code in this pane and click the <strong>Run button</strong> to verify that the rule runs as intended. A <strong>matches panel</strong> appears after Semgrep is run to display matches and tests.</dd>
+    <dt>Top menu</dt>
+    <dd>Save, share, and add your rule to the Rule Board through this menu.</dd>
+</dl>
+
+To **resize the panes**, position your mouse over the borders and click-drag to the desired width. The library pane can be completely hidden to give more space for the editing panes.
+
+Semgrep Registry rules are initially grouped by **directory**. Most of these directories correspond to languages. The Library can also be grouped by **rulesets**, which are rules arranged by category, such as security, best practices, and frameworks.
+
+To group by ruleset, right click on the empty space on the **official registry** entry and select **Group by ruleset**.
+
+![editor-groupbyruleset.png](../img/editor-groupbyruleset.png)
+
+## Creating a rule
+
+To create a rule, click **Create rule** on the splash page or the **(+) sign** next to the Library label. The Editor provides the minimum fields to create a rule:
+
+- `id`
+- `pattern`
+- `message`
+- `languages`
+- `severity`
+
+Refer to [Rule syntax](https://semgrep.dev/docs/writing-rules/rule-syntax/) for all possible fields and values to create a rule.
+
+### Jumpstart rule writing using existing rules
+
+Another method of creating rules is by **forking/copying** from existing rules found in Semgrep Registry. For example, Semgrep’s Java `crypto` ruleset prohibits the use of weak hashing algorithms `SHA-1` and `MD5`. An organization may want to further limit the use of other hash functions as part of their standards or security compliance. The next steps illustrate forking through copying an existing `use-of-sha1` rule and changing the rule to forbid MD2 hashes.
+
+1. Enter related terms into the search bar to find potential rules. In our example, this is `SHA1.`
+![Screenshot of Library pane with SHA1 filter](../img/editor-SHA1.png)
+
+2. Registry rules cannot be edited directly. They can be **forked** by **right-clicking on their entry** in the Library and selecting **Fork rule.**
+![Screenshot of forking menu](../img/editor-forking.png)
```

Comment:
This screenshot could also use a soft border and maybe some margin or padding around it, since there is written text very close to it and the screenshot itself includes text as well.

Reply:
Will do. (And also noting for future exquisiteness when we get there.)